src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/PhysicalAddress.cs

```diff
@@ -125,6 +127,32 @@ public static PhysicalAddress Parse(string address)
                     throw new FormatException(SR.Format(SR.net_bad_mac_address, address));
                 }
             }
+            else if (address.Contains(':'))
+            {
+                hasColons = true;
+                int segments = address.Split(':').Length;
```

Comment:
This pattern is going to allocate a string array and then discard it only keeping the count. Is there a way to do this with fewer or no allocations?

Reply:
Yes, we should avoid this.  It'd be cheaper to just iterate through the address counting the number of `:`s.